docs/proposals/buildstrategy-steps-resources.md

```diff
@@ -0,0 +1,306 @@
+The flaw is that the resources numbers, apply to all of the steps(_containers_) listed on the related Build/ClusterBuild Strategy, hoarding resources. This ends up with a higher number for resources consumption than what the user initially defined in the `Build` instance. For example, if users define `500m` as CPU request, but the strategy consists of five steps, then the used
+resources will be `500m` multiplied by `5`. This of course, can have implications for billing.
+
+Also, with the current `spec.resources` API we have in the `Build`, we are constrain to only define a single set of values for resources (_e.g. memory, cpu, etc_). This is not flexible while
+there could be situations where containers under the same strategy required different resource values (_not sharing the same values_). This raise the following questions:
+
+- First, why do we have in the `Build`/`BuildRun` an API to define `resources` for containers, if the end-users neither know the number of containers for a particular strategy, nor have insights into what each of these containers is doing.
+
+- Second, if a strategy consists of five steps(_containers_) and because of X reason I need to define five different values for each container CPU, how could I achieve this via the current `spec.build` API. (_not flexible_)
+
+## Motivation
+
+For strategies with multiple steps like Buildpacks, not all the steps(_containers_) will require the same set of resources(_e.g. memory, cpu, etc_) values. 
+In order to be able to manage situations where setting particular steps resources is required, we need a more flexible way to deal with N number of steps under the same strategy (_Build/ClusterBuildStrategy_). Also, we still need to help users to know that independently of the abstraction of the strategies, there are options for speeding up the builds. This options could be presented in the form of different `flavours` of the same strategy(_cluster or namespaced scope_), that differ between each other in terms of container resources values.
+
+### Goals
+
+- Keep abstracting the `end-user` from the strategies content.
+
+- Allow `strategy admins` to set different flavours of the **same** strategies. This is the `strategy admin` responsability, and these flavours may vary depending on the Cloud provider. Recommendations
+of different flavours for each strategy could be made, based on the steps and the notion around, which of them consume higher numbers for cpu/memory.
+
+- Set `default` strategies flavours in this repository, so that all strategy containers have limits on the resources they consume, avoiding OOM and CPU throttling when running in Travis.
+
+- Modify the `Build` and the `BuildRun` API, to remove the `spec.resources` configuration path.
+
+- Ensure that the `resources` defined in each `_Build/ClusterBuildStrategy_` will actually be processed and passed on runtime to the generated `TaskRun` from Tekton.
```

Comment:
This isn't really a goal, rather a way to implement the proposal.

Reply:
indeed, let me update